### PR TITLE
[CALCITE-6712] FROM_BASE64 and PARSE_URL's return data type should al…

### PR DESCRIPTION
**Pull request by NobiGo**
…ways be nullable

### CODE REVIEW

**Review by mihaibudiu — COMMENTED** [review]
Can you also update the spec in the site/_docs/reference.md file?

**Comment by NobiGo** [comment]
> Can you also update the spec in the site/_docs/reference.md file?

Reference.md file mainly describes the function functionality, without adding any explanation about the return type of the function, I think we should not add it.

**Comment by mihaibudiu** [comment]
Not the type, but the fact that on illegal inputs the result is null

**Comment by NobiGo** [comment]
> Not the type, but the fact that on illegal inputs the result is null

Get it.

File: testkit/src/main/java/org/apache/calcite/sql/test/SqlTests.java [hunk 1]
```diff
@@ -65,6 +65,15 @@ public abstract class SqlTests {
   public static final TypeChecker ANY_TYPE_CHECKER = (sql, type) -> {
   };
 
+  /**
+   * Checker which data type should be nullable.
```
Comment (mihaibudiu) [inline testkit/src/main/java/org/apache/calcite/sql/test/SqlTests.java:69]: I would rephrase this as "checker which enforces that a data type is nullable"

File: testkit/src/main/java/org/apache/calcite/sql/test/SqlOperatorFixture.java [hunk 2]
```diff
@@ -344,6 +344,18 @@ void checkString(
    */
   void checkNull(String expression);
 
+  /**
+   * Tests that a SQL expression returns the SQL NULL value and
+   * returns the data type is NULLABLE. For example,
```
Comment (mihaibudiu) [inline testkit/src/main/java/org/apache/calcite/sql/test/SqlOperatorFixture.java:349]: and the data type of the returned result is nullable

**Review by mihaibudiu — APPROVED** [review]
